File: backend/gnu/fapi/config/cors.php
<?php
/**
 * CORS 설정
 * .env 파일의 CORS_ALLOWED_ORIGINS 값을 쉼표로 구분하여 배열로 변환
 */
$allowedOrigins = getenv('CORS_ALLOWED_ORIGINS') ?: 'http://localhost:5173,http://localhost:3000';

return [
    'allowed_origins' => array_map('trim', explode(',', $allowedOrigins)),
    'allow_credentials' => getenv('CORS_ALLOW_CREDENTIALS') !== 'false', // 기본값: true
    'allowed_headers' => ['Content-Type', 'X-Requested-With', 'Authorization'],
    'allowed_methods' => ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS', 'PATCH'],
    'max_age' => (int)(getenv('CORS_MAX_AGE') ?: 86400),
];

File: backend/rust/api/src/middleware/cors.rs
use axum::{
    http::{HeaderValue, Method},
    response::Response,
};
use tower_http::cors::{Any, CorsLayer};

use crate::config::Config;
use std::sync::Arc;

pub fn create_cors_layer(config: &Arc<Config>) -> CorsLayer {
    let origins: Vec<HeaderValue> = config
        .cors_allowed_origins
        .iter()
        .filter_map(|origin| origin.parse().ok())
        .collect();

    CorsLayer::new()
        .allow_origin(origins)
        .allow_methods([
            Method::GET,
            Method::POST,
            Method::PUT,
            Method::DELETE,
            Method::OPTIONS,
        ])
        .allow_headers([
            "content-type".parse().unwrap(),
            "x-requested-with".parse().unwrap(),
            "authorization".parse().unwrap(),
        ])
        .allow_credentials(true)
        .max_age(std::time::Duration::from_secs(86400))
}
